wp-content/themes/currentorg/inc/business-directory-plugin.php

```diff
@@ -0,0 +1,204 @@
+				'depth'            => 1,
+				'echo'             => false,
+				'selected'         => $wpbdp_tag_parent_category,
+			)
+		),
+		__( 'Select the parent category for this tag.', 'currentorg' )
+	);
+
+	wp_nonce_field( 'wpbdp_tag_parent_category_update', 'wpbdp_tag_parent_category_nonce' );
+
+}
+add_action( 'wpbdp_tag_add_form_fields', 'wpbdp_tag_edit_form_fields' );
+add_action( 'wpbdp_tag_edit_form_fields', 'wpbdp_tag_edit_form_fields' );
+
+/**
+ * Save custom form fields from wpbdp_tag taxonomy add/edit pages
+ *
+ * @param int $term_id
+ * @param int $tt_id
+ */
+function wpbdp_tag_form_fields_save( $term_id, $tt_id ) {
+
+	if( isset( $_POST['wpbdp_tag_parent_category_nonce'] ) && wp_verify_nonce( $_POST['wpbdp_tag_parent_category_nonce'], 'wpbdp_tag_parent_category_update' ) ) {
+
+		if ( ! empty( $_POST['wpbdp_tag_parent_category'] ) ) {
+
+			update_term_meta( $term_id, 'wpbdp_tag_parent_category', 'wpbdp_category--' . $_POST['wpbdp_tag_parent_category'] );
+
+		}
+
+	}
+
+}
+add_action( 'created_wpbdp_tag', 'wpbdp_tag_form_fields_save', 10, 2 );
+add_action( 'edited_wpbdp_tag', 'wpbdp_tag_form_fields_save', 10, 2 );
+
+/**
+ * Modify the wpdb_tag tag cloud to include wpbdp_category--id class
+ *
+ * @param Array $tag_data
+ * @return Array the new tag data
+ */
+function wpbdp_modify_tag_cloud( $tag_data ) {
+	return array_map(
+		function ( $tag ) {
+			$term = get_term( $tag['id'] );
+
+			if ( 'wpbdp_tag' === $term->taxonomy ) {
+
+				$wpbdp_tag_parent_category = get_term_meta( $tag['id'] );
+				$wpbdp_tag_parent_category = $wpbdp_tag_parent_category['wpbdp_tag_parent_category'][0];
+				$tag['class'] .= ' ' . $wpbdp_tag_parent_category;
+
+			}
+
+			return $tag;
+		},
+		(array) $tag_data
+	);
+}
+add_filter( 'wp_generate_tag_cloud_data', 'wpbdp_modify_tag_cloud' );
+
+/**
+ * Return all tags in a list in the wpbdp_tag tag cloud.
+ *
+ * Runs during an AJAX call to get the tags in the list.
+ *
+ * @param Array $args
+ */
+function wpbdp_tag_cloud_show_all_tags( $args ) {
+	
+	if( in_array( 'wpbdp_tag', $args['taxonomy'] ) ){
+
+		if ( defined( 'DOING_AJAX' ) && DOING_AJAX && isset( $_POST['action'] ) && $_POST['action'] === 'get-tagcloud') {
+
+			unset( $args['number'] );
+			$args['hide_empty'] = 0;
+
+		}
+
+	}
+
+	return $args;
+}
+add_filter( 'get_terms_args', 'wpbdp_tag_cloud_show_all_tags' );
+
+/**
+ * Show all existing tags inside of the wpbdp_tag tag cloud
+ */
+function wpbdp_tag_cloud_custom_css_js(){
+
+	global $current_screen;
+
+	if( 'wpbdp_listing' === $current_screen->post_type ) {
+		?>
+		<script type="text/javascript">
+			jQuery(window).load(function() {
+
+				jQuery("body.wp-admin #tagsdiv-wpbdp_tag #link-wpbdp_tag").trigger("click");
+				jQuery("body.wp-admin #tagsdiv-wpbdp_tag #link-wpbdp_tag").hide();
+
+			});
+
+			jQuery(document).ready(function(){
+
+				// if a category is selected/deselected, do things
+				jQuery("#wpbdp_categorychecklist input").on("click", function(){
```

Comment:
The markup for the clickable UI is this:

```html
<li id="wpbdp_category-4195" class="popular-category">
    <label class="selectit">
        <input value="4195" type="checkbox" name="tax_input[wpbdp_category][]" id="in-wpbdp_category-4195" checked="checked">
            ::before
        </input>
        Consulting
    </label>
</li>
```

The `<input>` itself is hidden, and clicking on the label or the `::before` doesn't appear to toggle the visibility of the child terms.

Reply:
@benlk Is this issue still valid?

Reply:
This was resolved, yes.